README.md

```diff
@@ -1 +1,159 @@
+
+**Importantly**, we would like to give all candidates taking this test the same opportunity to solve the exercise in their own way. Because of this, **please do not fork or share this repo (nor your solution) with anybody 🙏🏻**
+
+<img align="center" src="https://i.imgur.com/ekyJNd9.jpg" width="600">
+
+## Instructions
+
+1. Push your solution to a **private repo** in your **personal Github account**
+2. When you are ready for us to take a look, please add [adjust-frontend-hiring][adjust-frontend-hiring] (GitHub user) as collaborator
+
+## General notes
+
+- The final solution must be a working React application.
+  - It's up to you how to bootstrap a new React application.
+- You can use any third-party library if you see a good fit for it.
+- It's not necessary to heavily comment the code, please only leave comments where they are necessary for code comprehension. However, we welcome notes and comments that reflect the higher-level decisions made during the challenge.
+- Aspects to be assessed:
+  - Code Style – meaningful naming conventions, consistent code formatting, and emphasis on code readability and future extensibility.
+  - Quality Assurance – the solution you upload should work as expected, conforming to the specifications of the tasks.
+  - Introducing best practices such as tests, type safety, and linting is appreciated.
+
+Good luck and happy coding! 🚀
+
+---
+
+## Exercise
+
+### TASK #1 - JavaScript
+
+Implement `prepareData` higher order function, which takes an object of filter params `{year, customerName}`, and returns a function that processes a list of missions, by only showing the ones that were launched in `year` and carried a payload belonging to `customerName`.
+
+**Observations:**
+
+- Missions should appear in inverse chronological order (sort), with the exception of those that carried more payloads should appear first.
+- Payloads are carried in the second stage of a rocket and they can belong to multiple customers.
+- It doesn't matter to which `customerName` 'program' each payload belongs to as long as `customerName` is the customer.
+
+**Example:**
+
+Considering we have the list of missions from this [data fixture][data-fixtures], and the following filter params:
+
+```js
+{
+  year: 2018,
+  customerName: "NASA"
+}
+```
+
+The expected result should be:
+
+```js
+[
+  {
+    flight_number: 62,
+    mission_name: "Iridium NEXT Mission 6",
+    payloads_count: 2,
+  },
+  {
+    flight_number: 72,
+    mission_name: "CRS-16",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 64,
+    mission_name: "CRS-15",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 60,
+    mission_name: "TESS",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 59,
+    mission_name: "CRS-14",
+    payloads_count: 1,
+  },
+];
+```
```

Comment:
Maybe does it make sense to create a table instead:

| Input | Expected |
| --- | --- |
| {} | {} |

WDYT?

Reply:
I tried to render the JSONs in the table format, it wasn't rendered neatly with GitHub markdown, unfortunately. I think it's ok to leave it as is.